apps/condo/domains/acquiring/schema/RegisterMultiPaymentForVirtualReceiptService.js

```diff
@@ -185,6 +189,17 @@ const RegisterMultiPaymentForVirtualReceiptService = new GQLCustomSchema('Regist
                     implicitFee: String(implicitFee),
                     serviceFee: String(fromReceiptAmountFee),
                 }
+
+                const amountToPay = Big(amount)
+                    .add(Big(paymentCommissionFields.explicitServiceCharge))
+                    .add(Big(paymentCommissionFields.explicitFee))
+                if (acquiringIntegration.minimumPaymentAmount && Big(amountToPay).lt(acquiringIntegration.minimumPaymentAmount)) {
+                    throw new GQLError({
+                        ...ERRORS.PAYMENT_AMOUNT_LESS_THAN_MINIMUM,
+                        messageInterpolation: { minimumPaymentAmount: Big(acquiringIntegration.minimumPaymentAmount).toFixed(2) },
+                    }, context)
+                }
+
```

Comment:
Looks like this logic is common for multiple modules, do you think we can move this away to some sort of utility function or up the callstack?

Reply:
Maybe to the hooks of the `MultiPayment` model

Reply:
Before multiPayment is created, several Payments are created. We can not check the total amount of MultiPayment inside Payment model. So, we can not implement hook to Payment model and if we first create payments and then throw error in MultiPayment - all created Payments will be useless